tmk_core/common/chibios/eeprom_stm32.h

```diff
@@ -61,18 +67,26 @@
 #        define FEE_MCU_FLASH_SIZE 384  // Size in Kb
 #    elif defined(MCU_STM32F303CC)
 #        define FEE_MCU_FLASH_SIZE 256  // Size in Kb
+#    elif defined(MCU_STM32F411xE)
+#        define FEE_MCU_FLASH_SIZE 512  // Size in Kb
 #    else
 #        error "No MCU type specified. Add something like -DMCU_STM32F103RB to your compiler arguments (probably in a Makefile)."
 #    endif
 #endif
 
 // DONT CHANGE
 // Choose location for the first EEPROM Page address on the top of flash
-#define FEE_PAGE_BASE_ADDRESS ((uint32_t)(0x8000000 + FEE_MCU_FLASH_SIZE * 1024 - FEE_DENSITY_PAGES * FEE_PAGE_SIZE))
-#define FEE_DENSITY_BYTES ((FEE_PAGE_SIZE / 2) * FEE_DENSITY_PAGES - 1)
-#define FEE_LAST_PAGE_ADDRESS (FEE_PAGE_BASE_ADDRESS + (FEE_PAGE_SIZE * FEE_DENSITY_PAGES))
+#define FEE_PAGE_END_ADDRESS ((uint32_t)(0x08000000 + FEE_MCU_FLASH_SIZE * 1024))
+#define FEE_PAGE_BASE_ADDRESS ((uint32_t)((FEE_PAGE_END_ADDRESS) - FEE_DENSITY_PAGES * FEE_PAGE_SIZE))
+#define FEE_DENSITY_BYTES ((FEE_PAGE_SIZE / 4) * FEE_DENSITY_PAGES - 1) // 4 Bytes for address, value pair
+//#define FEE_LAST_PAGE_ADDRESS (FEE_PAGE_BASE_ADDRESS + (FEE_PAGE_SIZE * FEE_DENSITY_PAGES))
+#define FEE_EMPTY_BYTE ((uint8_t)0xFF)
 #define FEE_EMPTY_WORD ((uint16_t)0xFFFF)
-#define FEE_ADDR_OFFSET(Address) (Address * 2)  // 1Byte per Word will be saved to preserve Flash
+#define FEE_EMPTY_VALUE ((uint32_t)0xFFFFFFFF)
+//#define FEE_ADDR_OFFSET(Address) (Address * 2)      // 1 Byte per Word will be saved to preserve Flash
+#define FEE_DATA_ADDRESS(Address) ((Address)+2)     // flash address of the eeprom data
+#define FEE_ADDR_ADDRESS(Address) (Address)         // flash address of the eeprom address
+#define FEE_MAX_BYTES ((FEE_DENSITY_BYTES+1) > 1024 ? 1024 : (FEE_DENSITY_BYTES+1)) // eeprom size
```

Comment:
I think this is a bug, it should be

```
#define FEE_MAX_BYTES ((FEE_DENSITY_BYTES+1) > 1024 ? (FEE_DENSITY_BYTES+1) : 1024)
```

Reply:
The Atmega32u4's eeprom size was 1024, so it was limited to keep the compatibility.

Reply:
Hmm, so you do not want to use the full space? Supposedly there is no problem if you use a larger external EEPROM.

Reply:
If VIA didn't exist, I would say that is just fine.  But ... it does, so a higher max by default is absolutely beneficial. To the point that I think it's more important than compatibility. 

In fact, for the emulated eeprom for the F303, isn't it 4kb?